common4j/src/main/com/microsoft/identity/common/java/cache/MsalOAuth2TokenCache.java

```diff
@@ -693,16 +694,28 @@ public ICacheRecord load(@NonNull final String clientId,
         final List<Credential> allCredentials = mAccountCredentialCache.getCredentials();
 
         // Load the AccessTokens
-        final List<Credential> accessTokens = mAccountCredentialCache.getCredentialsFilteredBy(
+        List<Credential> accessTokens;
+        if (authScheme instanceof PopAuthenticationSchemeWithClientKeyInternal) {
+            accessTokens = mAccountCredentialCache.getPoPAccessTokensFilteredBy(
```

Comment:
it's probably okay to just add the two params to getCredentialsFilteredBy(), so that we don't have to branch out the code here.

The method name is sort of misleading as well.... since the normal PoP scheme won't be going through this code path.